### PR TITLE
Port for Elasticsearch 2.x.

### PR DESCRIPTION
**Pull request by jjhuang**
Address issues I found in https://github.com/ChristopherRabotin/bungiesearch/issues/148. I figured since I already saw most if it that I could give it a shot or at least whittle down parts of the work for you.

Sorry if it's kind of messy -- very open to feedback. We think this library is great and would love to see it work with ES2.x.

### CODE REVIEW

File: bungiesearch/indices.py [hunk 1]
```diff
@@ -68,11 +68,14 @@ def matches_indexing_condition(self, item):
...
-    def get_mapping(self):
+    def get_mapping(self, meta_fields=True):
         '''
+        Returns the mapping for the index as a dictionary.
+
+        :param meta_fields: Also include elasticsearch meta fields in the dictionary.
         :return: a dictionary which can be used to generate the elasticsearch index mapping for this doctype.
         '''
-        return {'properties': dict((name, field.json()) for name, field in iteritems(self.fields))}
+        return {'properties': dict((name, field.json()) for name, field in iteritems(self.fields) if meta_fields or name not in AbstractField.meta_fields)}
```
Comment (jjhuang) [inline bungiesearch/indices.py]: There may be a cleaner/better way to do the meta field exclusion.

File: tests/core/models.py [hunk 2]
```diff
@@ -24,10 +24,11 @@ class Article(models.Model):
     class Meta:
         app_label = 'core'
 
+
 class User(models.Model):
     name = models.TextField(db_index=True)
     user_id = models.TextField(blank=True, primary_key=True)
-    description = models.TextField(blank=True)
+    about = models.TextField(blank=True)
```
Comment (jjhuang) [inline tests/core/models.py]: Field name change part of the conflict with `Article.description`.

File: setup.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 from setuptools import find_packages, setup
 
-VERSION = (1, 3, 0)
+VERSION = (2, 0, 0)
```
Comment (ChristopherRabotin) [inline setup.py]: I'll handle that part as the versioning of bungiesearch is not tied to the version of elasticsearch.

File: tests/core/models.py (hunk 2 above)
Comment (ChristopherRabotin) [inline tests/core/models.py]: What conflict are you talking about?

File: bungiesearch/indices.py (hunk 1 above)
Comment (ChristopherRabotin) [inline bungiesearch/indices.py]: I think this looks fine.

File: tests/core/models.py (hunk 2 above)
Comment (jjhuang) [inline tests/core/models.py]: `Article.description` uses a `snowball` analyzer whereas `User.description` is mapped to use an `edge_ngram_analyzer`, so ES complains that the mappings are different for fields with the same name.

File: setup.py (hunk 3 above)
Comment (jjhuang) [inline setup.py]: Ok I'll revert this part.

File: tests/core/models.py (hunk 2 above)
Comment (ChristopherRabotin) [inline tests/core/models.py]: Oh interesting. Is th

Comment (jjhuang) [inline tests/core/models.py]: Yep, they're really enforcing the field name within same type thing now.

> ### Conflicting field mappings
> Fields with the same name, in the same index, in different types, must have the same mapping, with the exception of the copy_to, dynamic, enabled, ignore_above, include_in_all, and properties parameters, which may have different settings per field.
(https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking_20_mapping_changes.html)

**Comment by NullSoldier** [comment]
Sorry but don't merge this until the other PR merges first. If you apply this on top of that one, it messes things up. I'll have jjhuang rebase this Pr on the other PR once it merges.

File: runtests.sh [hunk 4]
```diff
@@ -35,7 +35,7 @@ if ! is_responding $CLUSTER_URL; then
   else
     if [ "$1" != "--cluster" ]; then
       echo 'Starting single elasticsearch node'
-      $ES_PATH &> /dev/null &
+      $ES_PATH -D es.index.number_of_replicas=0 &> /dev/null &
```
Comment (jjhuang) [inline runtests.sh]: This can be removed when #150 is merged in.

File: runtests.sh [hunk 5]
```diff
@@ -44,6 +44,7 @@ if ! is_responding $CLUSTER_URL; then
         -D es.node.name="mycluster-node2" \
         -D es.node.master=true \
         -D es.node.data=false \
+        -D es.index.number_of_replicas=0 \
```
Comment (jjhuang) [inline runtests.sh]: Same.

**Comment by ChristopherRabotin** [comment]
@NullSoldier , your next PR is more than welcome ;)

File: runtests.sh (hunk 5 above)
Comment (ChristopherRabotin) [inline runtests.sh]: Okay, a new PR will have to be opened.

**Pull request merged**